sql/auth/sql_authentication.cc

```diff
@@ -3375,6 +3370,8 @@ int acl_authenticate(THD *thd, enum_server_command command) {
       DBUG_RETURN(1);
     }
 
+    thd->update_connection_certificate();
```

Comment:
Should this check for failure? If acl_check_ssl() returned 0, but then update_connection_certificate() hits a problem allocating memory during Bio_new, the certificate stored for the connection would be "". Would it be safer to fail the authentication?

Reply:
@hermanlee My reasoning is that this certificate string stored in THD is used only for information purposes and in my opinion in case of OOM it is better to provide inaccurate information but let users log in rather than completely deny their access. In any case, leaving this up to you to decide. It's not a big deal to rework this.